### ISSUE TITLE
Inconsistent results

### ISSUE THREAD

**Issue by plozmun** [netgen/ezplatform-site#64]
I have started to use this bundle with ezplatform-kernel 7 but it seems filter service don't work properly.

I hace the following controller and results differ from eZ Search Service. Whats wrong in it?
```
<?php

namespace AppBundle\Controller;

use eZ\Publish\API\Repository\Values\Content\Query\Criterion\ParentLocationId;
use Netgen\Bundle\EzPlatformSiteApiBundle\Controller\Controller;
use Netgen\Bundle\EzPlatformSiteApiBundle\View\ContentView;
use eZ\Publish\API\Repository\Values\Content\LocationQuery;


class ContentController extends Controller
{
    /**
     * @param \Netgen\Bundle\EzPlatformSiteApiBundle\View\ContentView $view
     *
     * @return \Netgen\Bundle\EzPlatformSiteApiBundle\View\ContentView
     */
    public function indexAction(ContentView $view)
    {
        $locationId = $view->getParameter('locationId');

        $filterService = $this->getSite()->getFilterService();

        $seachService = $this->get('ezpublish.api.service.search');

        $query = new LocationQuery(['filter' => new ParentLocationId($locationId)]);
        $searchResult = $filterService->filterLocations($query);

        // Total count: 0
        dump($searchResult->totalCount);

        $searchResult = $seachService->findContent($query);

        // Total count: 3
        dump($searchResult->totalCount);
        
        return $view;
    }
}

**Comment by emodric** [comment]
@plozmun Are you using Solr in your app? Filter service is not using Solr search engine to fetch the content, while eZ Search service does. This might be the cause of the discrepancy in results between filter service and search service. To verify if that is the case, you can try using find service from this package and see what results you get with it. Find service is basically the same thing as filter service, but uses Solr as a backend instead of the database.

If that doesn't help, one other reason could be is that filter and find services automatically filter content and locations based on prioritized list of languages configured for the current siteaccess, which, again, search service from eZ does not.

**Comment by plozmun** [comment]
No, i don't use Solr in the app. Then i can't use findService without Solr? With findService it works properly. 

I have overide parameter:
```
    netgen.ezplatform_site.prioritized_languages:
        - 'esl-ES'
        - 'eng-GB'
```
but this does't work either

**Comment by emodric** [comment]
You can use find service even if you don't use Solr. Find service actually falls back to legacy if no Solr is present. I'm not sure why it gives different results from filter service if you don't use Solr. @pspanja do you have a clue why this would happen?

BTW, you should not override the `netgen.ezplatform_site.prioritized_languages` parameter, it is used internally and not meant to be configured directly.

**Comment by pspanja** [comment]
@plozmun I've noticed in one case you use `filterLocations()` and in other `findContent()`.
Is there any change in behavior if you search for the same on both sides, `filterLocations()` and `findLocations()` or `filterContent()` and `findContent()`?

**Comment by plozmun** [comment]
I have try with filterContent and findContent and it`s the same.

It's strange because PermissionCriterionResolver returns false with filterService and true with ez Searchservice

**Comment by plozmun** [comment]
I realized it's our fault. Close this issue. Sorry for the inconvenience .

Thank you

**Comment by pspanja** [comment]
No worries @plozmun, good to know that you solved the problem 👍

**Comment by plozmun** [comment]
i reopen it , sorry 😆 

I dont know why the UserReference used by filter service is 
```
UserReference {#1467 ▼
  -userId: 10
}
```
and admin is:
```
UserReference {#854 ▼
  -userId: 14
}
```
Do you know why?

**Comment by pspanja** [comment]
That is coming from eZ Platform kernel, in default installation ID=10 is anonymous and ID=14 is admin. Are you testing that with different logins?

**Comment by plozmun** [comment]
Nope, 😢

**Comment by emodric** [comment]
@plozmun We have identified the issue, however, it will take a couple of days to solve it as it is not really straightforward, so please be patient :)

**Comment by pspanja** [comment]
Here we go: #67 

@plozmun your feedback would be much appreciated!

**Comment by pspanja** [comment]
Fixed by #67 

@plozmun thanks for reporting this!
